.github/workflows/cut-release.yml

```diff
@@ -0,0 +1,59 @@
+name: "cut-release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'version'
+        required: false
+jobs:
+  cut_release:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code 
+        uses: actions/checkout@v3
+      - name: "Build & test"
+        run: |
+          sudo apt-get install build-essential -y
+          ./autogen.sh
+          ./configure
+          make distcheck
+      - name: Update VERSION file on checkout 
+        run: |
+          set -x
+          sudo apt-get install -y python3-semver
+          if [ -z "$NEW_VERSION" ]; then
+             echo "Current Version: $(cat VERSION)"
+             pysemver bump patch $(cat VERSION) > VERSION
+          else
+             echo $NEW_VERSION > VERSION
+          fi
+          export NEW_VERSION=$(cat VERSION)
+          echo "::set-output name=NEW_VERSION::$NEW_VERSION" 
+        id: bump-version-file  
+        env:
+          NEW_VERSION: ${{ inputs.version }}
+      - name: Commit VERSION file 
```

Comment:
I think you're missing a step to push the commit (and tag?) to the master/main branch of the repository.

At least I looked at [your test releases](https://github.com/lrfurtado/numactl/releases) and they don't seem to be created on a commit that's bumping the version up. Can you please confirm that this works as expected?

Reply:
you are not seing them because I keep force pushing the latest squased version of my code.

Reply:
It’s a pity you started this PR with your “master” branch, since then your experiments end up showing up at the PR and you need to clean them up every time.

Consider renaming your “master” branch on your GitHub repo to another name. You can do that (while keeping this PR open) by using the following procedure:
https://github.blog/changelog/2021-01-19-support-for-renaming-an-existing-branch/

Then, you can use your repo’s “master” (or “main”) for experiments, and you can link to commits and tags there from this PR so that we can look at how it shows when you test it.

I’ll probably have some time at the computer this afternoon (on the phone right now) so I’ll take your PR for a spin.

Thanks for all your work so far! This is really useful!

Reply:
The problem was if the code was not on main/master the workflow would not get picked up for testing , but I will play with the branch names.

Reply:
Yes, sure. You can handle that by merging your feature branch into your repo’s master/main to test the feature.

Reply:
I checked this PR in [my repo](https://github.com/filbranden/numactl) and I still see that the release/tag is being created not on the new "Release 2.0.16" commit, but on the commit that precedes it.

See for example [my tags](https://github.com/filbranden/numactl/tags):

<img width="678" alt="Screen Shot 2022-09-25 at 12 25 49 AM" src="https://user-images.githubusercontent.com/106037/192133172-528e8ede-f9ff-4d3e-af51-4e04137d0fef.png">

Or [my releases](https://github.com/filbranden/numactl/releases) (after changing it from draft to published):

<img width="711" alt="Screen Shot 2022-09-25 at 12 27 12 AM" src="https://user-images.githubusercontent.com/106037/192133200-96ade629-6a1b-4909-b3ac-d6776681b275.png">

Looking at [my commits](https://github.com/filbranden/numactl/commits/main), it's clear that the release commit is being created (in my case, it's f63cf6f635eb972ff3ef0830e9fa7079ea880da3 on top of your 7d4f68bbc6e1634e036ef69c45a47902f013545a), but the tag is being created incorrectly.

The tarball looks good, it has the updated VERSION contents and unpacking and building it works as expected, so all is good on that front!

Can you please take one more look into why this is not picking up that last pushed commit when the tag is being created? Thanks!

Reply:
I've done some experimentation and didn't find a way to solve this by doing both in the same action.

I think you actually need to split this into two separate actions.

One to just update the VERSION file and create a new commit with an annotated tag and push that to the main branch on the repository.

Then another separate one that gets triggered when a tag is pushed, and then it does the build and creates the GitHub release. The second one has both the `make distcheck` part and the marvinpinto/action-automatic-releases action. See the ["when tags are pushed to repository"](https://github.com/marvinpinto/action-automatic-releases#create-a-new-github-release-when-tags-are-pushed-to-the-repository) example, but in short you probably want this as a trigger for the action that builds a tarball and creates a GitHub release:

```
on:
  push:
    tags:
      - "v*"
```